### PR TITLE
Ban classes from extending `Actor` and `DistributedActor` protocol explicitly

### PR DESCRIPTION
**Pull request by ktoso**
WIP, some cleanup pending.

### CODE REVIEW

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -4349,7 +4349,12 @@ ERROR(async_objc_dynamic_self,none,
       "asynchronous method returning 'Self' cannot be '@objc'", ())
 
 ERROR(actor_inheritance,none,
-      "actor types do not support inheritance", ())
+      "%select{actor|distributed actor}0 types do not support inheritance",
+      (bool))
+
+ERROR(actor_protocol_illegal_inheritance,none,
+      "illegal explicit inheritance from 'Actor' protocol, "
```
Comment (harlanhaskins) [inline include/swift/AST/DiagnosticsSema.def:4356]: `illegal` is not a term I've seen thrown around in diagnostics, can we pick something less jarring? 😄

How about `classes cannot conform to the 'Actor' protocol; declare this as %select{an actor|a distributed actor}0 instead`?

Comment (ktoso) [reply include/swift/AST/DiagnosticsSema.def:4356]: Thanks @harlanhaskins ! I was struggling with the right wording here.

@DougGregor suggested `Non-actor type %0 cannot conform to the 'Actor' protocol` how does that sound? 

And there's a fixit to change struct->actor etc :)

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS and merge

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Comment by ktoso** [comment]
Related to rdar://78975415

**Comment by ktoso** [comment]
resolves rdar://79749091

**Pull request merged**